core/embed/rust/src/ui/display.rs

```diff
@@ -227,6 +748,42 @@ impl Font {
     pub fn line_height(self) -> i32 {
         constant::LINE_SPACE + self.text_height()
     }
+
+    pub fn get_glyph(self, ch: char) -> Option<Glyph> {
+        let gl_data = display::get_char_glyph(ch, self.0);
+
+        if gl_data.is_null() {
+            return None;
+        }
+
+        unsafe {
+            let width = *gl_data.offset(0) as i32;
+            let height = *gl_data.offset(1) as i32;
+            let adv = *gl_data.offset(2) as i32;
+            let bearing_x = *gl_data.offset(3) as i32;
+            let bearing_y = *gl_data.offset(4) as i32;
+            let data = gl_data.offset(5);
+            Some(Glyph::new(width, height, adv, bearing_x, bearing_y, data))
+        }
+    }
+
+    pub fn display_text(
+        self,
+        text: &'static str,
+        baseline: Point,
+        fg_color: Color,
+        bg_color: Color,
+    ) {
+        let colortable = get_color_table(fg_color, bg_color);
+        let mut adv_total = 0;
+        for c in text.chars() {
+            let g = self.get_glyph(c);
+            if let Some(gly) = g {
+                let adv = gly.print(baseline + Offset::new(adv_total, 0), colortable);
+                adv_total += adv;
+            }
```

Comment:
The C `display_get_glyph` function returns nonprintable glyph if it finds that the character is for example unicode, so that glyph is printed. `display_get_glyph` returns null for successive bytes of unicode characters, which we do not want to print. (since in rust the encoding is checked before calling `display_get_glyph` the successive bytes should be discarded before actually calling `display_get_glyph`). Therefore the else branch seems redundant